core/src/avm2/e4x.rs

```diff
@@ -715,3 +715,30 @@ pub fn to_xml_string<'gc>(
     to_xml_string_inner(xml, &mut buf)?;
     Ok(AvmString::new(activation.context.gc_context, buf))
 }
+
+pub fn name_to_multiname<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    name: &Value<'gc>,
+) -> Result<Multiname<'gc>, Error<'gc>> {
+    if let Value::Object(o) = name {
+        if let Some(qname) = o.as_qname_object() {
+            return Ok(qname.name().clone());
+        }
+    }
+
+    let name = name.coerce_to_string(activation)?;
+
+    if name.starts_with(b'@') {
```

Comment:
nit: `name.strip_prefix('@')` can be used here, avoiding the slice `name[1..]` right after.

Reply:
Nice. I didn't know of this method before.